src/DGtal/geometry/surfaces/estimation/PlaneProbingParallelepipedEstimator.h

```diff
@@ -0,0 +1,337 @@
+
+          /**
+           * A wrapper around the "is a point inside the plane?" predicate.
+           *
+           * @param aPoint any digital point.
+           * @return true if the point is inside, false otherwise.
+           */
+          bool InPlane (Point const& aPoint) const;
+
+          /**
+           * The NotAbove predicate, see @cite LMRJMIV2020.
+           *
+           * @param aPoint any digital point.
+           * @return true if the point is not above the fixed point, false otherwise.
+           */
+          bool operator() (Point const& aPoint) const;
+
+      private:
+          const Predicate* myPredicate = nullptr; /**< A pointer to the InPlane predicate */
+          Integer myBound; /**< The bound. */
+          Self* myParallelpipedEstimator = nullptr; /**< A pointer to a plane-probing parallelepiped estimator. */
+
+          /**
+           * Returns the fixed point 'q'.
+           */
+          Point q () const;
+      };
+
+
+    // ----------------------- Standard services ------------------------------
+  public:
+    /**
+     * Default constructor.
+     */
+    PlaneProbingParallelepipedEstimator() = delete;
+
+    /**
+     * Constructs a plane probing parallelepiped estimator from an initial frame and a probing predicate.
+     *
+     * @param aPoint the base point of the initial frame.
+     * @param aM the three vectors of the initial frame.
+     * @param aPredicate the probing predicate.
+     * @param aBound the bound used in the NotAbove predicate.
+     */
+    PlaneProbingParallelepipedEstimator(Point const& aPoint, Triangle const& aM,
+                                        Predicate const& aPredicate, Integer const& aBound);
+
+    /**
+     * Destructor.
+     */
+    ~PlaneProbingParallelepipedEstimator() = default;
+
+    /**
+     * Copy constructor.
+     * @param other the object to clone.
+     */
+    PlaneProbingParallelepipedEstimator ( const PlaneProbingParallelepipedEstimator & other ) = delete;
+
+    /**
+     * Move constructor.
+     * @param other the object to move.
+     */
+    PlaneProbingParallelepipedEstimator ( PlaneProbingParallelepipedEstimator && other ) = delete;
+
+    /**
+     * Copy assignment operator.
+     * @param other the object to copy.
+     * @return a reference on 'this'.
+     */
+    PlaneProbingParallelepipedEstimator & operator= ( const PlaneProbingParallelepipedEstimator & other ) = delete;
+
+    /**
+     * Move assignment operator.
+     * @param other the object to move.
+     * @return a reference on 'this'.
+     */
+    PlaneProbingParallelepipedEstimator & operator= ( PlaneProbingParallelepipedEstimator && other ) = delete;
+
+    // ----------------------- Plane Probing services ------------------------------
+  public:
+    /**
+     * @param aIndex the index of the vector (between 0 and 2).
+     * @return the i-th height vector \f$ m_i \f$.
+     */
+    Point m (int aIndex) const;
```

Comment:
I wonder if defining a Vector type (same as Point) would not clarify the returned value.

Reply:
I agree, I also changed it in the tetrahedron version.